ips/src/tests.rs

```diff
@@ -200,12 +219,73 @@ fn destroy_should_fail() {
         assert_eq!(
             IpsStorage::<Runtime>::get(0),
             Some(IpsInfoOf::<Runtime> {
-                owner: primitives::utils::multi_account_id::<Runtime, <Runtime as Config>::IpsId>(
-                    0
-                ),
+                parentage: Parentage::Parent(primitives::utils::multi_account_id::<
+                    Runtime,
+                    <Runtime as Config>::IpsId,
+                >(0, None)),
                 metadata: MOCK_METADATA.to_vec().try_into().unwrap(),
-                data: vec![0]
+                data: vec![AnyId::IpfId(0)].try_into().unwrap(),
+                allow_replica: true,
+                ips_type: IpsType::Normal,
             })
         );
     });
 }
+
+#[test]
+fn create_replica_should_work() {
+    ExtBuilder::default().build().execute_with(|| {
+        assert_ok!(Ipf::mint(
+            Origin::signed(ALICE),
+            MOCK_METADATA.to_vec(),
+            H256::from(MOCK_DATA)
+        ));
+
+        assert_eq!(Ips::next_ips_id(), 0);
+        assert_ok!(Ips::create_ips(
+            Origin::signed(ALICE),
+            MOCK_METADATA.to_vec(),
+            vec![0],
+            true,
+        ));
+
+        assert_eq!(Ips::next_ips_id(), 1);
+        assert_ok!(Ips::create_replica(Origin::signed(ALICE), 0));
+
+        let ips_0 = IpsStorage::<Runtime>::get(0).unwrap();
+        let ips_1 = IpsStorage::<Runtime>::get(1).unwrap();
+
+        assert_eq!(ips_0.data, ips_1.data);
+
+        assert_eq!(ips_0.metadata, ips_1.metadata);
+
+        assert_ne!(ips_0.parentage, ips_1.parentage);
```

Comment:
These assertions are incomplete, they don't test the whole ips
```suggestion
        assert_eq!(ips_1, IpsInfo {
            parentage: Parentage::Parent(multi_account_id(1, None)),
            metadata: ips_0.metadata,
            data: ips_0.data,
            ips_type: IpsType::Replica(0),
            allow_replica: false
        });
```